core/typedefs.h

```diff
@@ -71,6 +71,22 @@
 #endif
 #endif
 
+// No discard allows the compiler to flag warnings if we don't use the return value of functions / classes
+#ifndef _NO_DISCARD_
+// c++ 17 onwards
+#if __cplusplus >= 201703L
+#define _NO_DISCARD_ [[nodiscard]]
+#else
+#define _NO_DISCARD_
+#endif
+#endif
```

Comment:
The check is correct but since it's always true (Godot can't compile without C++17 or later, even if changed manually in `SConstruct`), we can simplify further:
```suggestion
#ifndef _NO_DISCARD_
#define _NO_DISCARD_ [[nodiscard]]
#endif
```

I agree about keeping the macro itself for now so that relevant changes are easy to cherry-pick to `3.x`.